include/Dialect/Btor/IR/BtorOps.td

```diff
@@ -838,13 +839,12 @@ def AssertNotOp : Btor_Op<"assert_not"> {
         ```
     }];
 
-    let arguments = (ins I1:$arg);
+    let arguments = (ins Btor_BitVec:$arg);
 
-    let assemblyFormat = "`(` $arg `)` attr-dict";
+    let assemblyFormat = "`(` $arg `)` attr-dict `:` type($arg)";
 }
-
-def ConstantOp : Btor_Op<"constant", [ConstantLike, NoSideEffect,
-    AllTypesMatch<["value", "result"]>]> {
+//,AllTypesMatch<["value", "result"]>
```

Comment:
You know enough now to have the types matching for the constant op. Uncomment this line - it makes it easier to verify the operation.